src/FormContainer.tsx

```diff
@@ -98,20 +105,34 @@ const makeWrapper = <T extends {}>(config: IFormConfig<T>) => (WrappedComponent:
             }
         });
 
+        handleSubmit = (submit: () => Promise<any>) => () => {
```

Comment:
should submit func take in values as arguments?

Reply:
Opted to inject form values in handleSubmit as suggested below.